Pod/Classes/LSBinaryCodec.m

```diff
@@ -0,0 +1,324 @@
+//
+//  LSBinaryCodec.m
+//
+
+#import "LSBinaryCodec.h"
+#import "LSSpanContext.h"
+#import "OTTracer.h"
+
+
+@implementation LSBinaryCodec
+
++ (NSData *)encodedMessageForTraceID:(UInt64)traceID
+                              spanID:(UInt64)spanID
+                             baggage:(NSDictionary *)baggage {
+    // Encode our inner message first
+    NSData *inner = [LSBinaryCodec encodeInnerMessageForTraceID:traceID spanID:spanID baggage:baggage];
+
+    // Next, write our outer message by checking the length of the inner message.
+    // TODO: We probably know this capacity, so we should set it:
+    NSMutableData *outer = [[NSMutableData alloc] init];
+
+    // BasicCarrier basic_ctx = 2;
+    [LSBinaryCodec writeTagNumber:2 format:WireFormatMessage buffer:outer];
+    [LSBinaryCodec writeVarintEncodedUInt64:inner.length buffer:outer];
+    [outer appendData:inner];
+
+    return outer;
+}
+
++ (BOOL)decodeMessage:(NSData *)protoEnc
```

Comment:
I don't think we need to separate them. I was just thinking about minimizing the surface area for this code entirely... i.e., it's not about the binary size or whether they get linked in, but just whether we need to maintain it.

I did have one thought overnight about this (ah, insomnia): we could skip baggage for v1 of this code (fine to keep it around in a git branch or whatever). Once all parties feel confident about the "flat" varint stuff, we could then introduce baggage and unittest the hell out of it, including intentionally corrupt data and so forth.